fairpy/items/fairly_allocating_few_queries.py

```diff
@@ -0,0 +1,360 @@
+    >>> Alice = fairpy.agents.AdditiveAgent({"a": 2, "b": 6, "c": 3, "d":1,"e":1,"f":7,"g":15}, name="Alice")
+    >>> George = fairpy.agents.AdditiveAgent({"a": 2, "b": 6, "c": 3, "d":1,"e":1,"f":7,"g":15}, name="George")
+    >>> allocation = two_agents_ef1([Alice,George],["a", "b", "c", "d", "e", "f", "g"])
+    >>> allocation
+    Alice gets {g} with value 15.
+    George gets {a,b,c,d,e,f} with value 20.
+    <BLANKLINE>
+    >>> Alice.is_EF1(allocation[0], allocation) and George.is_EF1(allocation[1], allocation)
+    True
+    >>> Alice = fairpy.agents.AdditiveAgent({"a": 2, "b": 6, "c": 3, "d":1,"e":4,"f":7,"g":15}, name="Alice")
+    >>> George = fairpy.agents.AdditiveAgent({"a": 2, "b": 6, "c": 3, "d":1,"e":4,"f":7,"g":15}, name="George")
+    >>> allocation = two_agents_ef1([Alice,George],["a", "b", "c", "d", "e", "f", "g"])
+    >>> allocation
+    Alice gets {f,g} with value 22.
+    George gets {a,b,c,d,e} with value 16.
+    <BLANKLINE>
+    >>> Alice.is_EF1(allocation[0], allocation) and George.is_EF1(allocation[1], allocation)
+    True
+    >>> ### ONLY ONE OBJECT
+    >>> Alice = fairpy.agents.AdditiveAgent({"a": 2}, name="Alice")
+    >>> George = fairpy.agents.AdditiveAgent({"a": 2}, name="George")
+    >>> allocation = two_agents_ef1([Alice,George],["a"])
+    >>> allocation
+    Alice gets {} with value 0.
+    George gets {a} with value 2.
+    <BLANKLINE>
+    >>> Alice.is_EF1(allocation[0], allocation) and George.is_EF1(allocation[1], allocation)
+    True
+    >>> ### nothing to allocate
+    >>> Alice = fairpy.agents.AdditiveAgent({}, name="Alice")
+    >>> George = fairpy.agents.AdditiveAgent({}, name="George")
+    >>> allocation = two_agents_ef1([Alice,George],[])
+    >>> allocation
+    Alice gets {} with value 0.
+    George gets {} with value 0.
+    <BLANKLINE>
+    >>> Alice.is_EF1(allocation[0], allocation) and George.is_EF1(allocation[1], allocation)
+    True
+    >>> Alice = fairpy.agents.AdditiveAgent({"a": 2, "b": 6, "c": 3, "d":1,"e":1,"f":7,"g":15,"h":21,"i":4,"j":23,"k":7,"l":10,"m":11,"n":22,"o":6,"p":9,"q":16,"r":12,"s":3,"t":28,"u":39,"v":2,"w":9,"x":1,"y":17,"z":100}, name="Alice")
+    >>> George = fairpy.agents.AdditiveAgent({"a": 2, "b": 6, "c": 3, "d":1,"e":1,"f":7,"g":15,"h":21,"i":4,"j":23,"k":7,"l":10,"m":11,"n":22,"o":6,"p":9,"q":16,"r":12,"s":3,"t":28,"u":39,"v":2,"w":9,"x":1,"y":17,"z":100}, name="George")
+    >>> allocation = two_agents_ef1([Alice,George],["a", "b", "c", "d", "e", "f", "g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"])
+    >>> allocation
+    Alice gets {t,u,v,w,x,y,z} with value 196.
+    George gets {a,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s} with value 179.
+    <BLANKLINE>
+    >>> Alice.is_EF1(allocation[0], allocation) and George.is_EF1(allocation[1], allocation)
+    True
+    """
+    if len(agents) != 2:
+        raise Exception("wrong number of agents")
+    if _are_identical_valuations(agents) is False:
```

Comment:
האם האלגוריתם לשני שחקנים גם דורש העדפות זהות?

Reply:
אני לא בטוח לגבי זה, עכשיו כשאני חושב על זה די הגיוני שלא, במאמר כתוב כך: "allocation for two agents with arbitrary monotonic valuations".
בכל מקרה להתאים את זה לא יקח לי הרבה זמן

Reply:
האם דחפת את השינויים? אני עדיין רואה את הגירסה הישנה.

Reply:
עכשיו כן